blockauth/views/tests/test_oauth_views.py

```
@@ -740,3 +740,157 @@ def test_first_oauth_signup_populates_first_name_when_field_exists(self, db):
+# ---------------------------------------------------------------------------
+#
+# Each web callback view must forward the provider's first/last name as
+# `extra_user_fields` to `SocialIdentityService.upsert_and_link`, so the
+# user model's `first_name` / `last_name` are populated on signup rather
+# than NULL. The test user model here (`tests.TestBlockUser`) extends
+# `AbstractBaseUser` and does not declare those fields, so we mock the
+# service to assert the kwarg payload — the integration the view is
+# responsible for.
+
+
+def _stub_upsert_returning(user):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_stub_upsert_returning`

(ANN202)